macros/shared

```diff
@@ -152,6 +152,25 @@ CROSS_COMPILATION_CONF_EOF\
 %cross_meson_install \
   %ninja_install -C %{_cross_vpath_builddir}
 
+%_cross_cmake_toolchain_conf %{_cross_vpath_srcdir}/toolchain.%{_cross_target}.cmake
+
+%write_cross_cmake_toolchain_conf \
+cat <<CROSS_CMAKE_TOOLCHAIN_EOF > %_cross_cmake_toolchain_conf 2>/dev/null || :\
+set(CMAKE_SYSTEM_NAME Linux)\
+set(CMAKE_SYSTEM_PROCESSOR %{_cross_arch})\
+set(triple %{_cross_triple})\
+set(CMAKE_SYSROOT %{_cross_rootdir})\
+set(CMAKE_C_COMPILER %{_cross_target}-gcc)\
+set(CMAKE_C_COMPILER_TARGET %_cross_target)\
+set(CMAKE_CXX_COMPILER %{_cross_target}-g++)\
+set(CMAKE_CXX_COMPILER_TARGET %_cross_target)\
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)\
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)\
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)\
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)\
+CROSS_CMAKE_TOOLCHAIN_EOF\
+%{nil}
```

Comment:
We only have one user of `%cross_cmake` - the `docker-init` package - so I'd prefer to go ahead and convert that macro to generate and use the toolchain file, the way that `%cross_meson` does.